client/src/utils.js

```diff
@@ -197,6 +197,19 @@ export default {
       arr &&
       arr.filter(n => !isNaN(parseFloat(n)) && isFinite(n)).map(n => Number(n))
     )
+  },
+
+  preventNegativeAndLongDigits: function(valueStr, maxLen) {
+    const nonDigitsRemoved = valueStr.replace(/[^0-9]+/g, "")
+    let safeValue = nonDigitsRemoved
+    if (maxLen > 0) {
+      safeValue =
+        nonDigitsRemoved.length > maxLen
+          ? nonDigitsRemoved.slice(0, maxLen)
+          : nonDigitsRemoved
+    }
+
+    return safeValue
```

Comment:
Code can be simplified to:
```
    const nonDigitsRemoved = valueStr.replace(/\D/g, "")
    return maxLen <= 0 ? nonDigitsRemoved : nonDigitsRemoved.slice(0, maxLen)
```

Reply:
Function changed with negatives forced to 0